pvlib/pvsystem.py

```diff
@@ -1952,11 +1952,144 @@ def v_from_i(resistance_shunt, resistance_series, nNsVth, current,
+        thermal voltage under the desired IV curve conditions (Vth). The
+        thermal voltage of the cell (in volts) may be calculated as
+        ``k*temp_cell/q``, where k is Boltzmann's constant (J/K),
+        temp_cell is the temperature of the p-n junction in Kelvin, and
+        q is the charge of an electron (coulombs).
+        0 < nNsVth
+
+    voltage : numeric
+        The voltage in Volts under desired IV curve conditions.
+
+    saturation_current : numeric
+        Diode saturation current in amperes under desired IV curve
+        conditions. Often abbreviated ``I_0``.
+        0 < saturation_current
+
+    photocurrent : numeric
+        Light-generated current (photocurrent) in amperes under desired
+        IV curve conditions. Often abbreviated ``I_L``.
+        0 <= photocurrent
+
+    Returns
+    -------
+    current : np.ndarray or scalar
+
+    References
+    ----------
+    [1] A. Jain, A. Kapoor, "Exact analytical solutions of the
+    parameters of real solar cells using Lambert W-function", Solar
+    Energy Materials and Solar Cells, 81 (2004) 269-277.
+    '''
+    try:
+        from scipy.optimize import brentq
+    except ImportError:
+        raise ImportError('This function requires scipy')
+
+    # This transforms Gsh=1/Rsh, including ideal Rsh=np.inf into Gsh=0., which
+    #  is generally more numerically stable
+    conductance_shunt = 1./resistance_shunt
+
+    # Check for all scalar or rank-0 inputs, any transform for consistent indexing
+    if all(map(lambda x: np.isscalar(x) or (isinstance(x, np.ndarray) and not x.shape), [conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent])):
+        all_scalar_or_rank0_inputs = True
+        rank0_output = any(map(lambda x: isinstance(x, np.ndarray), [conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent]))
+        scalar_output = not rank0_output
+        # Make sure that we can index numpy arrays
+        conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent = map(lambda x: np.array([x]), [conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent])
+    else:    
+        all_scalar_or_rank0_inputs = False
+        rank0_output = False
+        scalar_output = False
+        # Make sure that we can index numpy arrays
+        conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent = np.broadcast_arrays(conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent)
+
+    # Compute initially with zero r_s_Ohm
+    current_ic = photocurrent - saturation_current*np.expm1(voltage/nNsVth) - conductance_shunt*voltage
+    
+    f = lambda current, conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent: current_sum_diode_node(voltage, current, conductance_shunt, resistance_series, nNsVth, saturation_current, photocurrent)
```

Comment:
IMO if you should online use lambda for an anonymous inline function, if you are going to write it on it's own line, then you should write it as a function instead so either do this:
```
def f(lambda current, conductance_shunt, resistance_series, nNsVth, voltage, saturation_current, photocurrent): 
    return current_sum_diode_node(voltage, current, conductance_shunt, resistance_series, nNsVth, saturation_current, photocurrent)
```
or move this down to where you use f on line 2075.

see [pep 8](http://pep8.org/#programming-recommendations)

Reply:
Also would be lovely to see these lines wrapped down to < 80 characters so I can read them without scrolling. thanks!

Reply:
This code is for illustration only, so please go easy on the over-conciseness with my lambda functions. I did add many comments :). That said, I'm sorry the formatting isn't better: I haven't figured out yet how to set up VSCode to auto format and for personal projects I just let my editor wrap my code to the window on my 14" laptop.